src/lightning_app/runners/cloud.py

```diff
@@ -305,6 +292,15 @@ def dispatch(
 
             if find_instances_resp.lightningapps:
                 existing_instance = find_instances_resp.lightningapps[0]
+
+                # TODO: support multiple instances / 1 instance per cluster
+                if existing_instance.spec.cluster_id != cluster_id:
```

Comment:
@luca3rd I am confused about this. If the app already exists on a cluster, than it means the user is going to restart it when a new release.